lib/semantics/mod-file.cc

```diff
@@ -371,7 +374,8 @@ bool ModFileReader::Read(const SourceName &modName) {
     return false;
   }
   auto &modSymbol{*it->second};
-  modSymbol.scope()->set_chars(parsing.cooked().MoveChars());
+  // TODO: Preserve the CookedSource rather than a copy of its string.
+  modSymbol.scope()->set_chars(std::string{parsing.cooked().data()});
```

Comment:
Reading module files won't work until this TODO is resolved. The CharBlocks will have references to the old cooked characters, not the saved ones.

Reply:
fixing